graphql_compiler/schema_transformation/split_query.py

```diff
@@ -0,0 +1,642 @@
+        the out_name of the @output directive uniquely identifying the field used or stitching
+        in this sub query node
+    """
+    child_type_name, child_selections = _get_child_type_and_selections(ast, type_info)
+    # Get existing field with name in child
+    child_property_field = _get_property_field(child_selections, child_field_name, [])
+    # Add @output if needed, record out_name
+    child_output_name = _get_out_name_optionally_add_output(
+        child_property_field, intermediate_out_name_assigner
+    )
+    # Get new child_selections
+    child_selections = _replace_or_insert_property_field(child_selections, child_property_field)
+    # Wrap around
+    child_query_ast = _get_query_document(child_type_name, child_selections)
+    child_query_node = SubQueryNode(child_query_ast)
+
+    return child_query_node, child_output_name
+
+
+def _get_property_field(selections, field_name, directives_from_edge):
+    """Return a Field object with field_name, sharing directives with any such existing field.
+
+    Any valid directives in directives_on_edge will be transferred over to the new field.
+    If there is an existing Field in selection with field_name, the returned new Field
+    will also contain all directives of the existing field with that name.
+
+    Args:
+        selections: List[Union[Field, InlineFragment]]. If there is a field with field_name,
+                    the directives of this field will carry over to the output field. It is
+                    not modified by this function
+        field_name: str, the name of the output field
+        directives_from_edge: List[Directive], the directives of a vertex field. The output
+                              field will contain all @filter and any @optional directives
+                              from this list
+
+    Returns:
+        Field object, with field_name as its name, containing directives from any field in the
+        input selections with the same name and directives from the input list of directives
+    """
+    new_field = ast_types.Field(
+        name=ast_types.Name(value=field_name),
+        directives=[],
+    )
+
+    # Check parent_selection for existing field of given name
+    parent_field = try_get_ast(selections, field_name, ast_types.Field)
+    if parent_field is not None:
+        # Existing field, add all its directives
+        directives_from_existing_field = parent_field.directives
+        if directives_from_existing_field is not None:
+            new_field.directives.extend(directives_from_existing_field)
+
+    # Transfer directives from edge
+    if directives_from_edge is not None:
+        for directive in directives_from_edge:
+            if directive.name.value == u'output':  # output is illegal on edge field
+                raise GraphQLValidationError(
+                    u'Directive "{}" is not allowed on an edge field, as @output directives '
```

Comment:
I believe these are called `vertex field` in the spec: https://github.com/kensho-technologies/graphql-compiler#definitions